internal/app/referenceclient/impl.go

```diff
@@ -117,8 +118,21 @@ func (i *invoker) unary(
 	var trailers []*v1.Header
 	payloads := make([]*v1.ConformancePayload, 0, 1)
 
+	ctx, wire := WithWireCapture(ctx)
```

Comment:
I think a more intuitive abstraction for this would be something like so:
```go
ctx := withWireCapture(ctx)

// perform RPC using ctx

// then at the very end
wireDetails := getWireDetails(ctx)
// ... do stuff with wire details
```
That way the wire wrapper never needs to be exposed and doesn't need any methods -- it's just an implementation detail of your wire tracer.

Even if you keep the shape the way it is, I think this means you don't need to provide a `Set` method -- only `Get` needs to be exposed.